src/assets/TokenRatesController.ts

```diff
@@ -312,29 +336,90 @@ export class TokenRatesController extends BaseController<
 
-    const newContractExchangeRates: ContractExchangeRates = {};
+    let newContractExchangeRates: ContractExchangeRates = {};
     if (!slug) {
       this.tokenList.forEach((token) => {
         const address = toChecksumHexAddress(token.address);
         newContractExchangeRates[address] = undefined;
       });
     } else {
-      const pairs = this.tokenList.map((token) => token.address).join(',');
-      const query = `contract_addresses=${pairs}&vs_currencies=${nativeCurrency.toLowerCase()}`;
-      const prices = await this.fetchExchangeRate(slug, query);
+      const { nativeCurrency } = this.config;
+      newContractExchangeRates = await this.fetchAndMapExchangeRates(
+        nativeCurrency,
+        slug,
+      );
+    }
+    this.update({ contractExchangeRates: newContractExchangeRates });
+  }
+
+  /**
+   * Checks if the active network's native currency is supported by the coingecko API
+   * If supported. fetches and maps contractExchange rates in format to be consumed by UI
+   * If not supported fetches contractExchange rates and maps from from token/fallback-currency to token/nativeCurrency
+   *
+   * @param nativeCurrency - the native currency of the currently active network
+   * @param slug - the unique slug used to id the chain by the coingecko api
+   * should be used to query token exchange rates
+   * @returns An object with conversion rates for each token
+   * related to the network's native currency
+   */
+  async fetchAndMapExchangeRates(
+    nativeCurrency: string,
+    slug: string,
+  ): Promise<ContractExchangeRates> {
+    const contractExchangeRates: ContractExchangeRates = {};
+
+    // check if native currency is supported as a vs_currency by the API
+    const nativeCurrencySupported = await this.checkIsSupportedVsCurrency(
+      nativeCurrency,
+    );
+
+    if (nativeCurrencySupported) {
+      // if it is we can do a simple fetch against the coingetck API
+      const prices = await this.fetchExchangeRate(slug, nativeCurrency);
       this.tokenList.forEach((token) => {
-        const address = toChecksumHexAddress(token.address);
         const price = prices[token.address.toLowerCase()];
-        newContractExchangeRates[address] = price
+        contractExchangeRates[toChecksumHexAddress(token.address)] = price
           ? price[nativeCurrency.toLowerCase()]
           : 0;
       });
+    } else {
+      // if native currency is not supported we need to use a fallback vsCurrency, get the exchange rates
+      // in token/fallback-currency format and convert them to expected token/nativeCurrency format.
+      let tokenExchangeRates;
+      let nativeCurrencyConversionRate = 0;
+      try {
+        [
+          tokenExchangeRates,
+          { conversionRate: nativeCurrencyConversionRate },
+        ] = await Promise.all([
+          this.fetchExchangeRate(slug, FALL_BACK_VS_CURRENCY),
+          fetchNativeExchangeRate(FALL_BACK_VS_CURRENCY, nativeCurrency, false),
```

Comment:
Why are we using CryptoCompare here, as opposed to CoinGecko? :thinking: Do they have better support for converting ETH to the native currencies of custom networks?

Reply:
I defaulted to using them because they are what we use for all native currency conversions at the moment... Looking at the coingecko API looks like if we wanted to use Coingecko we would need to keep an additional cached (huge) [list of coins](https://www.coingecko.com/api/documentations/v3#/coins/get_coins_list) and then loop through and get their coin id to do the ETH price comparison.

Reply:
Oh, I thought we had the "coin ID" already. I thought it was the same thing as the "chain slug".

Reply:
No, unfortunately that's [a different endpoint/list ](https://www.coingecko.com/api/documentations/v3#/asset_platforms/get_asset_platforms)

Reply:
Sorry I didn't mean we had the entire list. I meant we have the "coin ID" for each custom network that CoinGecko supports (we get it via the `asset_platforms` call, it's the "id" property).

Reply:
I may be misunderstanding something but I don't think that's true. For instance, for polygon, in the coin/list I see
```
{
    "id": "matic-network",
    "symbol": "matic",
    "name": "Polygon"
  },
```
and for asset/platforms I see: 
```
  {
    "id": "polygon-pos",
    "chain_identifier": 137,
    "name": "Polygon POS",
    "shortname": "MATIC"
  },
  ```

Reply:
meanwhile for `BSC` for assets/platforms:
```
{
    "id": "binance-smart-chain",
    "chain_identifier": 56,
    "name": "Binance Smart Chain",
    "shortname": "BSC"
  },
  ```
  and in the coinlist there are a bunch of different coins like
  ```
   {
    "id": "benscoin",
    "symbol": "bsc",
    "name": "Benscoin"
  },
  ``` 
  with a symbol of BSC so we couldn't match `shortname` to `symbol` case insensitively as the polygon examples appears

Reply:
Darn! You're right, I was mistaken. Well that's awfully inconvenient.

I brought this up because now we're depending on two APIs here instead of one, so our "supported chains" state isn't correct anymore. Now our supported chains are (the ones that CoinGeck supports) AND (the ones that CoinGecko supports as vs currencies OR those that CryptoCompare supports).

I'll think on this for a bit, to try and see what the impact of this is.

Reply:
Well that's not quite how I would put it. We still only support the chains that CoinGecko supports because ultimately the token exchange rates come from a query that involves the chainSlug and the tokenAddresses of the tokens for that chain.  The reason this second step involving cryptoCompare is required is because weirdly though some of these chains are `supported platforms` (meaning we can use them to query for token rates of that chain) for Coingecko, they don't support the chain's native currency as `vsCurrency` for the token exchange rate endpoint (`simple/token_price/${chainSlug}?${query}`).

Reply:
>(the ones that CoinGeck supports) AND (the ones that CoinGecko supports as vs currencies OR those that CryptoCompare supports).

This might have been confusing. What I meant is:
>const chainSupported = supportedByCoinGecko(chainID) && (supportedAsCoinGeckoVsCurrency(chainId) || supportedByCryptoCompare(chainId))

Which is consistent with everything you just said.

Reply:
yes I guess I did understand that to be what you meant. But then
>"supported chains" state isn't correct anymore
doesn't quite make sense to me since before this PR that list contains many chains we didn't support and, in fact, it still does since many don't have a `chain_identifier` (what we call chainId).

Reply:
I was thinking of the supported chains state as being a representation of the chains we support, and I guess that's not true. The point I was trying to make was that by relying on two platforms, we reduce the number of chains we can support, and we increase the chances of users encountering problems due to API downtime or symbol confusion.

The symbol issue is scary. It seems like CryptoCompare doesn't list the `chainId` for the coins they support anywhere - they just accept the symbol. I can't find any API endpoint that says what each symbol represents. And we're relying solely upon the user to enter that symbol correctly already today in production (for the fiat conversions). So... not that this PR is introducing that reliance, but, it will mean the token prices could be displayed incorrectly now as well because of this intermediate step.

Reply:
> The point I was trying to make was that by relying on two platforms, we reduce the number of chains we can support

Perhaps the "can" here is what you mean to emphasize, because in terms of this PR there is no change for chains we supported before. 

I agree that the symbol issue isn't ideal. And as you acknowledge this reliance (on the accuracy of user input native currency symbols) is already there - and, when a user misenters this symbol, already may receive inaccurate conversion rates for the native currency on a custom network. I can think of some possible solutions to mitigate this issue while we wait for the coingecko api to improve (or a better one to crop up) - We could create an API or list that pre-populates the symbol ticker for chainId we know. But ultimately - though this deepens potential confusion when a user misenters a native currency symbol - I think this PR is an improvement for the short/medium term for the vast majority of users.

Reply:
Yeah, fair point. It's not great but I guess it's an improvement. I've left a comment about this in our shared Slack channel with CoinGecko but I won't block on this for now. Hopefully they come back with a solution for us, or hopeful news of future API improvements.